c++/src/Reader.hh

```diff
@@ -88,10 +88,12 @@ namespace orc {
     // Select a field by id
     void updateSelectedByFieldId(std::vector<bool>& selectedColumns, uint64_t fieldId);
     // Select a type by id
-    void updateSelectedByTypeId(std::vector<bool>& selectedColumns, uint64_t typeId);
+    void updateSelectedByTypeId(std::vector<bool>& selectedColumns, uint64_t typeId,
+                                const RowReaderOptions::IdReadIntentMap& idReadIntentMap);
 
     // Select all of the recursive children of the given type.
-    void selectChildren(std::vector<bool>& selectedColumns, const Type& type);
+    void selectChildren(std::vector<bool>& selectedColumns, const Type& type,
+                        const RowReaderOptions::IdReadIntentMap& idReadIntentMap);
```

Comment:
Actually, this doesn't match with the function description.
> // Select all of the recursive children of the given type.

Reply:
Clarified the comment as follow
```
    // Select a type id of the given type.
    // This function may also select all of the recursive children of the given type
    // depending on the read intent of that type in idReadIntentMap.
    void selectChildren(std::vector<bool>& selectedColumns, const Type& type,
                        const RowReaderOptions::IdReadIntentMap& idReadIntentMap);
```